packages/core/src/components/popover/popover.md

```diff
@@ -71,3 +157,244 @@ Modifiers are the tools through which you customize Popper.js's behavior. Popper
+prop.
+
+@### Sizing
+
+Popovers by default have a `max-width` but no `max-height`. To constrain the height of a popover
+and make its content scrollable, set the appropriate CSS rules on `.pt-popover-content`:
+
+```css.scss
+// pass "my-popover" to `popoverClassName` prop.
+.my-popover .pt-popover-content {
+    max-height: $pt-grid-size * 15;
+    overflow-y: auto;
+}
+```
+
+@reactExample PopoverSizingExample
+
+@### Minimal style
+
+You can create a minimal popover with the `pt-minimal` modifier: `popoverClassName="pt-minimal"`.
+This removes the arrow from the popover and makes the transitions more subtle.
+
+@reactExample PopoverMinimalExample
+
+This minimal style is recommended for popovers that are not triggered by an obvious action like the
+user clicking or hovering over something. For example, a minimal popover is useful for making
+typeahead menus where the menu appears almost instantly after the user starts typing.
+
+Minimal popovers are also useful for context menus that require quick enter and leave animations to
+support fast workflows. You can see an example in the [context menus](#core/components/context-menu)
+documentation.
+
+@## Testing
+
+<div class="pt-callout pt-intent-primary pt-icon-info-sign">
+    Your best resource for strategies in popover testing is
+    [its own unit test suite.](https://github.com/palantir/blueprint/blob/master/packages/core/test/popover/popoverTests.tsx)
+</div>
+
+#### Animation delays
+
+`Popover` can be difficult to test because it uses `Portal` to inject its contents elsewhere in the
+DOM (outside the usual flow); this can be simplified by using `inline` Popovers in tests.
+Hover interactions can also be tricky due to delays and transitions; this can be resolved by
+zeroing the default hover delays.
+
+```tsx
+ <Popover inline {...yourProps} hoverCloseDelay={0} hoverOpenDelay={0}>
+    {yourTarget}
+</Popover>
+```
+
+#### Rendering delays
+
+`Popover` delays rendering updates triggered on `mouseleave`, because the mouse might have moved from the popover to the target, which may require special handling depending on the current [`interactionKind`](http://localhost:9000/#core/components/popover.opening-and-closing). Popper.js also throttles rendering updates to improve performance. If your components are not updating in a synchronous fashion as expected, you may need to introduce a `setTimeout` to wait for asynchronous Popover rendering to catch up:
+
+```tsx
+import { Classes, Overlay, Popover, PopoverInteractionKind } from "@blueprintjs/core";
+import { assert } from "chai";
+import { mount } from "enzyme";
+import { Target } from "react-popper";
+
+wrapper = mount(
+    <Popover inline={true} interactionKind={PopoverInteractionKind.HOVER}>
+        <div>Target</div>
+        <div>Content</div>
+    </Popover>
+);
+
+wrapper.find(Target).simulate("mouseenter")
+wrapper.findClass(Classes.POPOVER).simulate("mouseenter")
+wrapper.findClass(Classes.POPOVER).simulate("mouseleave");
```

Comment:
😢 `findClass` is a method I wrote for the popover suite--it's not a part of enzyme. please refactor.